addons/website/static/src/snippets/s_cta_mobile/000.scss

```diff
@@ -0,0 +1,4 @@
+// This code forces the generated image to fit it's parent size and allow an overlap effect
+.o_grid_item_image img[data-shape='web_editor/devices/galaxy_front_portrait_half'] {
+    object-fit: fill !important;
+}
```

Comment:
1. I am not sure this is necessary, this gives this result when resizing:
![image](https://github.com/user-attachments/assets/15bfb06a-8254-499d-b3a1-a98835c0f2b0)

instead of 

![image](https://github.com/user-attachments/assets/49596b4f-34e6-4225-83c2-2c6e9a6e016c)

Neither is good, but the one without your code is better IMO.

2. You are adding this in s_cta_mobile/000.scss like if it is only for that snippet but this rules targets everything

Reply:
I agree it looks better without when you resize it but at the same time, when this code is not present, the default rendering of the image+shape looks like this:
| rendering | svg |
|--------|--------|
| <img width="1337" alt="Screenshot 2024-11-27 at 09 08 45" src="https://github.com/user-attachments/assets/bb5eb51e-5420-4b4d-8ca7-39814542c341"> | <img width="1277" alt="Screenshot 2024-11-27 at 09 38 31" src="https://github.com/user-attachments/assets/f9952d67-4972-429c-8d4c-2a55291236fe"> | 

Then, if I simply open the shapes catalog and hover one of them, it looks like a computation is done and the shape looks good again.
| rendering | svg |
|--------|--------|
| <img width="1003" alt="Screenshot 2024-11-27 at 09 43 24" src="https://github.com/user-attachments/assets/63de2491-58ad-4ec1-bef2-96a161177197"> | <img width="1279" alt="Screenshot 2024-11-27 at 09 38 41" src="https://github.com/user-attachments/assets/fc1a3050-b73d-4388-8621-f45b39952a6f"> | 

Honestly, all of this is a bit mystical for me. Do you have an idea on how to force this computation so the generated image is correct by default ? I'll push a version without this SCSS so you can have a look on how it behaves.

Reply:
@bvr-odoo Looks like the `/web_editor/image_shape` route is not computing this shape as nicely as the client side when using the image shape option. Could you have a look once you can?

Reply:
> @bvr-odoo Looks like the `/web_editor/image_shape` route is not computing this shape as nicely as the client side when using the image shape option. Could you have a look once you can?

@qsm-odoo Indeed, the SVG `data-forced-size` attribute (see [commit](https://github.com/odoo/odoo/commit/63cf8a693a5262600d465d70f0383229464d71e7)), used to update the SVG height, is not taken into account in the `/web_editor/image_shape route`. We need to add a check for `data-forced-size` in this route and adapt the SVG height as it is done on JS side. I can push a commit to this PR or fix it in a new PR in master (or stable version? not sure if it's worth it).

Reply:
Depends on the fix but it should be quite safe? Probably stable, no?

Reply:
> Depends on the fix but it should be quite safe? Probably stable, no?

I think it's safe yes, I create a PR in stable and maybe we have to wait to merge it to merge this one

Reply:
@qsm-odoo here is the PR https://github.com/odoo/odoo/pull/189156
Tested on v16.0 and on this branch, it works well
could take a look and merge it please

Reply:
Hello @anso-odoo 
Now that https://github.com/odoo/odoo/pull/189668 has been merged, could you please rebase this PR on Master and check if everything is okay? It would be great to merge it today so it has a chance to be frozen next week 😉 
cc @qsm-odoo

Reply:
Hello @bvr-odoo, @anso-odoo is off today, I just rebased and applied the changes requested [here](https://github.com/odoo/odoo/pull/185635#pullrequestreview-2465205672), I'll check if everything's okay while runbot is building 🙂

Reply:
@chgo-odoo I'll adapt what remains to be adapted and merge, don't worry about this no more but thanks :+1: